### PR TITLE
Use gatsby link in OS vs. Cloud guide

### PR DESCRIPTION
**Pull request by pfd**
## Description of the change

- uses Gatsby link for internal links
- copy edit
- use unicode for tables   ❌   ✅ 

Preview: 
- https://deploy-preview-586--rudderstack-docs-staging.netlify.app/get-started/cloud-vs-open-source/
- https://deploy-preview-586--rudderstack-docs-staging.netlify.app/get-started/rudderstack-architecture/

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Maintenance

## Related issues

n/a

### Development

- [x] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: docs/get-started/cloud-vs-open-source.mdx [hunk 1]
```diff
@@ -7,22 +7,22 @@ description: >-
...
+This guide compares the features in the two primary RudderStack offerings— [RudderStack Cloud](https://app.rudderstack.com/signup?type=freetrial) and [RudderStack Open Source](https://app.rudderstack.com/signup?type=opensource). It will help you choose the option that best suits your requirements.
 
 <div class="infoBlock">
 
-RudderStack also offers three plans as a part of <a href="https://app.rudderstack.com/signup?type=freetrial">RudderStack Cloud</a> : Free, Pro, and Enterprise. Refer to the  <a href="https://www.rudderstack.com/pricing/">Pricing</a> page for detailed comparison and additional information on these plans.
+RudderStack offers three plans as a part of <a href="https://app.rudderstack.com/signup?type=freetrial">RudderStack Cloud</a>: *Free*, *Pro*, and *Enterprise*. Refer to the  [Pricing](https://www.rudderstack.com/pricing) page for a detailed comparison and additional information on these plans.
```
Comment (ishachopra0401) [inline docs/get-started/cloud-vs-open-source.mdx:14]: **For linking within the RS doc (including info, success or warning blocks):**
`<Link to="rel-link”>Text</Link>`

**For linking any external webpage:**
`[Text](abs-link)`

**For linking any external webpage in the info, success or warning blocks:**
Do we use `<a href=“abs-link”>Text</a>` OR `[Text](abs-link)`? I see both these formats in the corresponding info block.

Comment (pfd) [reply docs/get-started/cloud-vs-open-source.mdx:14]: We should standardize on `[Text](abs-link)` for external links - The idea being that we should avoid using raw HTML whenever we can, and use Markdown instead. I was only focused on fixing internal links in this PR so good catch- i'll push fixes to the external ones that are left

Comment (ishachopra0401) [reply docs/get-started/cloud-vs-open-source.mdx:14]: Got it, makes sense!

**Review by ameypv-rudder — APPROVED** [review]

**Review by ishachopra0401 — APPROVED** [review]

**Comment by ameypv-rudder** [comment]
Updated the Get Started landing page and the Glossary as well so that we can wrap this section up with the updated links.

**Pull request merged**